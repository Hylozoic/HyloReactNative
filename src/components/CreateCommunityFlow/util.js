export const slugValidatorRegex = /^[0-9a-z-]{2,40}$/
export const invalidSlugMessage = 'URLs must have between 2 and 40 characters, and can only have lower case letters, numbers, and dashes.'
