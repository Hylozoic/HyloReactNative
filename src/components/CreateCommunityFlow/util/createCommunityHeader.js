import React from 'react'
import {
  Text
} from 'react-native'
import { HeaderBackButton } from 'react-navigation'
import { caribbeanGreen, white, white60onCaribbeanGreen } from 'style/colors'

export default function createCommunityHeader (title, navigation) {
  return {
    title: <Text style={styles.headerTitle}>{title}</Text>,
    color: white,
    fontSize: 10,
    headerStyle: {
      backgroundColor: caribbeanGreen,
      borderBottomWidth: 0,
      shadowRadius: 0,
      shadowOffset: {
        height: 0
      },
      elevation: 0
    },
    headerBackgroundColor: caribbeanGreen,
    headerLeft: <HeaderBackButton onPress={() => navigation.goBack()} tintColor={white60onCaribbeanGreen} />
  }
}

const styles = {
  headerTitle: {
    color: white,
    fontSize: 12,
    fontWeight: 'bold'
  }
}
