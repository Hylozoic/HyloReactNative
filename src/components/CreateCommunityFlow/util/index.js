export const slugValidatorRegex = /^[0-9a-z-]{2,40}$/
export const invalidSlugMessage = 'URLs must have between 2 and 40 characters, and can only have lower case letters, numbers, and dashes.'

export function formatDomainWithUrl (communityDomain) {
  if (!communityDomain) return
  let formattedDomain = communityDomain.replace(/hylo\.com\/c\/?/, '')
  if (formattedDomain !== '') {
    formattedDomain = 'hylo.com/c/' + formattedDomain
  }
  return formattedDomain
}

export function removeUrlFromDomain (communityDomain) {
  if (!communityDomain) return
  return communityDomain.replace('hylo.com/c/', '')
}
