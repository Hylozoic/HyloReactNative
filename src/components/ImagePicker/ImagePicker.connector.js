import { connect } from 'react-redux'
import apiHost from 'util/apiHost'
import Request from './Request'

const UPLOAD = 'UPLOAD'

export function upload (type, id, file) {
  const url = apiHost + '/noo/upload'

  const request = Request.create(url, 'POST')
  .set('type', type)
  .set('id', id)
  .set('file', file)

  return {
    type: UPLOAD,
    payload: request.send()
    .then(resp => {
      let { status, text } = resp
      if (status === 200) return JSON.parse(text)

      const error = new Error(text)
      error.response = {status}
      throw error
    })
  }
}

export default connect(null, {upload}, null, {withRef: true})
