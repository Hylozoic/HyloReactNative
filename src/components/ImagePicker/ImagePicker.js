import React, { Component } from 'react'
import { Text, TouchableOpacity, View, Image } from 'react-native'
import RNImagePicker from 'react-native-image-picker'
import { AWS_S3_BUCKET } from 'react-native-dotenv'
import aws from 'aws-sdk'

export default class ImagePicker extends Component {

  constructor (props) {
    super(props)
    this.state = {}
  }

  showPicker () {
    const { title = 'Choose an image' } = this.props

    var options = {
      title,
      storageOptions: {
        skipBackup: true,
        path: 'images'
      }
    }

    RNImagePicker.showImagePicker(options, (response) => {
      console.log('Response = ', response)

      if (response.didCancel) {
        console.log('User cancelled image picker')
      } else if (response.error) {
        console.log('ImagePicker Error: ', response.error)
      } else {
        this.setState({source: {uri: response.uri}})
        const s3 = new aws.S3()
        const key = '/testuploads/' + response.uri
        const body = 'data:image/png;base64,' + response.data
        return s3.upload({
          Bucket: AWS_S3_BUCKET,
          ACL: 'public-read',
          ContentType: 'image/png',
          Key: key,
          Body: body
        }).promise()
        .then(result => {
          console.log('s3 came back with', result)
        })
      }
    })
  }

  render () {
    const { source } = this.state

    return <View>
      <TouchableOpacity onPress={() => this.showPicker()}>
        <Text>Image Picker</Text>
        {!!source && <Image source={source} style={imageStyle}/>}
      </TouchableOpacity>
    </View>
  }
}

const imageStyle = {
  width: 256,
  height: 256
}
