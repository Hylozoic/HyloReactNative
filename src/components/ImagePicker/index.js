import component from './ImagePicker'
import connector from './ImagePicker.connector'

export default connector(component)
