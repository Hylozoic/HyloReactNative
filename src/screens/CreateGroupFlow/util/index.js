export const slugValidatorRegex = /^[0-9a-z-]{2,40}$/
export const invalidSlugMessage = 'URLs must have between 2 and 40 characters, and can only have lower case letters, numbers, and dashes.'

export function formatDomainWithUrl (groupDomain) {
  if (!groupDomain) return
  let formattedDomain = groupDomain.replace(/hylo\.com\/groups\/?/, '')
  if (formattedDomain !== '') {
    formattedDomain = 'hylo.com/groups/' + formattedDomain
  }
  return formattedDomain
}

export function removeDomainFromURL (groupDomain) {
  if (!groupDomain) return
  return groupDomain.replace('hylo.com/groups/', '')
}
