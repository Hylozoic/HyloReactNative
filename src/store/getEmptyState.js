import orm from './models'
import { combinedReducers } from './reducers'

export default function getEmptyState () {
  return combinedReducers({ orm: orm.getEmptyState() }, { type: '' })
}
