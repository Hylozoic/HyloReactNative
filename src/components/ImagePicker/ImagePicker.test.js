import React from 'react'
import { Provider } from 'react-redux'
import getEmptyState from 'store/getEmptyState'
import { render } from '@testing-library/react-native'
import { createMockStore } from 'util/testing'
import ImagePicker from 'components/ImagePicker'

jest.mock('react-native/Libraries/ActionSheetIOS/ActionSheetIOS', () => ({
  showActionSheetWithOptions: jest.fn()
}))

// jest.mock('components/ImagePicker/ImagePicker', () => ({
//   __esModule: true,
//   ...jest.requireActual('components/ImagePicker/ImagePicker'),
//   showImagePicker: jest.fn()
// }))

describe('ImagePicker Specification', () => {
  it('default render matches snapshot', async () => {
    const state = getEmptyState()

    const { toJSON } = render(
      <Provider store={createMockStore(state)}>
        <ImagePicker disabled={false} />
      </Provider>
    )
    expect(await toJSON()).toMatchSnapshot()
  })
})
