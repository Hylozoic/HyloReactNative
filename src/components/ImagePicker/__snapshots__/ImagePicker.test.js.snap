// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches last snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  disabled={true}
  onPress={[Function]}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Children
  </Text>
</TouchableOpacity>
`;
